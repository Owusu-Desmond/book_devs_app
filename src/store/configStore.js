import { combineReducers, configureStore } from '@reduxjs/toolkit';
import reservationsReducer from './reservations/reservations';
import authReducer from './user/auth';
import devReducer from './developers/devs/devReducer';
import addReservations from './reservations/addReservations';
import developersDetailsReducer from './developersDetails/developersDetails';

const rootReducer = combineReducers({
  reservations: reservationsReducer,
  authentication: authReducer,
  developer: devReducer,
  developerDetails: developersDetailsReducer,
  addReservations,
});

const store = configureStore({
  reducer: rootReducer,
});

export default store;
