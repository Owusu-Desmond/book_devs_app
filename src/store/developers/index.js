export { getDevelopers, addDev } from './devs/devActions';
