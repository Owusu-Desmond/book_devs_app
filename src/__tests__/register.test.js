import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Register from '../components/authentication/register';
import store from '../store/configStore';

describe('Test Register page', () => {
  const tree = render(
    <Provider store={store}>
      <BrowserRouter>
        <Register />
      </BrowserRouter>
    </Provider>,
  );
  it('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  it('Register page has Submit button', () => {
    expect(tree.findByText('Submit')).toBeTruthy();
  });

  it('Sign up page has a log in link', () => {
    expect(tree.findByText('Sign in')).toBeTruthy();
  });
});
