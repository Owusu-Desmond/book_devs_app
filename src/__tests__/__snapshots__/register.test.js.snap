// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Register page renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="Auth-form-container"
      >
        <form
          class="Auth-form"
        >
          <div
            class="Auth-form-content"
          >
            <h3
              class="Auth-form-title"
            >
              Sign Up
            </h3>
            <div
              class="text-center"
            >
              Already registered?
               
              <span
                class="link-primary"
              >
                <a
                  class=""
                  href="/login"
                >
                  Sign In
                </a>
              </span>
            </div>
            <div
              class="form-group mt-3"
            >
              <label
                for="fullname"
              >
                Full Name
              </label>
              <input
                class="form-control mt-1"
                id="fullname"
                placeholder="e.g Jane Doe"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group mt-3"
            >
              <label
                for="email"
              >
                Email address
              </label>
              <input
                class="form-control mt-1"
                id="email"
                placeholder="Email Address"
                required=""
                type="email"
                value=""
              />
            </div>
            <div
              class="form-group mt-3"
            >
              <label
                for="password"
              >
                Password
              </label>
              <input
                class="form-control mt-1"
                id="password"
                placeholder="Password"
                required=""
                type="password"
                value=""
              />
            </div>
            <div
              class="form-group mt-3"
            >
              <label
                for="confirm_password"
              >
                Confirm Password
              </label>
              <input
                class="form-control mt-1"
                id="confirm_password"
                placeholder="Confirm Password"
                required=""
                type="password"
                value=""
              />
            </div>
            <div
              class="d-grid gap-2 mt-3"
            >
              <button
                class="btn btn-primary"
                type="submit"
              >
                Submit
              </button>
            </div>
          </div>
        </form>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="Auth-form-container"
    >
      <form
        class="Auth-form"
      >
        <div
          class="Auth-form-content"
        >
          <h3
            class="Auth-form-title"
          >
            Sign Up
          </h3>
          <div
            class="text-center"
          >
            Already registered?
             
            <span
              class="link-primary"
            >
              <a
                class=""
                href="/login"
              >
                Sign In
              </a>
            </span>
          </div>
          <div
            class="form-group mt-3"
          >
            <label
              for="fullname"
            >
              Full Name
            </label>
            <input
              class="form-control mt-1"
              id="fullname"
              placeholder="e.g Jane Doe"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group mt-3"
          >
            <label
              for="email"
            >
              Email address
            </label>
            <input
              class="form-control mt-1"
              id="email"
              placeholder="Email Address"
              required=""
              type="email"
              value=""
            />
          </div>
          <div
            class="form-group mt-3"
          >
            <label
              for="password"
            >
              Password
            </label>
            <input
              class="form-control mt-1"
              id="password"
              placeholder="Password"
              required=""
              type="password"
              value=""
            />
          </div>
          <div
            class="form-group mt-3"
          >
            <label
              for="confirm_password"
            >
              Confirm Password
            </label>
            <input
              class="form-control mt-1"
              id="confirm_password"
              placeholder="Confirm Password"
              required=""
              type="password"
              value=""
            />
          </div>
          <div
            class="d-grid gap-2 mt-3"
          >
            <button
              class="btn btn-primary"
              type="submit"
            >
              Submit
            </button>
          </div>
        </div>
      </form>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
